.github/workflows/deploy-to-gcp.yaml

```diff
@@ -1,16 +1,20 @@
 name: Deploy Xplorers API to GCP
-
-on: push
+on: 
+  push:
+  pull_request:
 
 permissions:
-  contents: read
   id-token: write
+  contents: write
+  pull-requests: write
+  repository-projects: write
```

Comment:
Not sure why we need `repository-projects`? Does this write to the project board?
<img width="1147" alt="image" src="https://github.com/user-attachments/assets/994f52a3-0c6e-4c95-b344-3caa33a60546">

Reply:
Made changes Thanks